astropy/coordinates/transformations.py

```diff
@@ -741,8 +744,104 @@ def __call__(self, fromcoord, toframe):
                 'should have been of type {1}'.format(res, self.tosys))
         return res
 
+class BaseAffineTransform(CoordinateTransform):
+    """
+    Base class for common functionality between the ``AffineTransform``-type
+    subclasses.
+    """
+
+    def _apply_transform(self, fromcoord, matrix, vectors):
```

Comment:
Move to `CartesianRepresentation.transform`? (since this may be discussion to keep, see larger comment in the astropy thread: https://github.com/astropy/astropy/pull/6218#issuecomment-308886358)